specs/language/introduction.tex

```diff
@@ -130,37 +192,38 @@
 documented in the \gls{dx} Specifications, and this document will not attempt to
 elaborate further.
 
-\Sec{Common Definitions}{Intro.Defs}
-
-\p The following definitions are consistent between \acrshort{hlsl} and the
-\gls{isoC} and \gls{isoCPP} specifications, however they are included here for
-reader convenience.
-
-\Sub{Diagnostic Message}{Intro.Defs.Diags}
-\p An implementation defined message belonging to a subset of the
-implementation's output messages which communicates diagnostic information to
-the user.
-
-\Sub{Ill-formed Program}{Intro.Defs.IllFormed}
-\p A program that is not well formed, for which the implementation is expected
-to return unsuccessfully and produce one or more diagnostic messages.
-
-\Sub{Implementation-defined Behavior}{Intro.Defs.ImpDef}
-\p Behavior of a well formed program and correct data which may vary by the
-implementation, and the implementation is expected to document the behavior.
-
-\Sub{Implementation Limits}{Intro.Defs.ImpLimits}
-\p Restrictions imposed upon programs by the implementation.
-
-\Sub{Undefined Behavior}{Intro.Defs.Undefined}
-
-\p Behavior of invalid program constructs or incorrect data which this standard
-imposes no requirements, or does not sufficiently detail.
-
-\Sub{Unspecified Behavior}{Intro.Defs.Unspecified}
-\p Behavior of a well formed program and correct data which may vary by the
-implementation, and the implementation is not expected to document the behavior.
-
-\Sub{Well-formed Program}{Intro.Defs.WellFormed}
-\p An HLSL program constructed according to the syntax rules, diagnosable
-semantic rules, and the One Definition Rule.
+\Sub{Memory Spaces}{Intro.Memory.Spaces}
+
+\p \acrshort{hlsl} programs manipulate data stored in five distinct memory
+spaces: thread, threadgroup, device, constant and host.
+
+\Par{Thread Memory}{Intro.Memory.Spaces.Thread}
+Thread memory is local to the \gls{lane}. It is the default memory space used to
+store local variables. Thread memory cannot be directly read from other threads
+without the use of intrinsics to synchronize execution and memory.
+
+\Par{\gls{threadgroup} Memory}{Intro.Memory.Spaces.Group}
+\gls{threadgroup} memory is denoted in \acrshort{hlsl} with the
+\texttt{groupshared} keyword. The underlying memory for any declaration
+annotated with \texttt{groupshared} is shared across an entire
+\gls{threadgroup}. Reads and writes to \gls{threadgroup} Memory, may occur in
+any order except as restricted by synchronization intrinsics.
+
+\Par{Device Memory}{Intro.Memory.Spaces.Device}
+Device memory is memory available to all \gls{lane}s executing on the device.
+This memory may be read or written to by multiple \gls{threadgroup}s that are
+executing concurrently. Reads and writes to device memory may occur in any order
+except as restricted by synchronization intrinsics. Some device memory may be
```

Comment:
"synchronization intrinsics" might preclude specific annotations like `globallycoherent` which I don't personally consider to be an "intrinsic"